rust/src/origin.rs

```diff
@@ -388,6 +397,8 @@ mod test {
     fn test_origin_roundtrip() -> Result<()> {
         let kf = kf_from_str(BASE)?;
         origin_validate_roundtrip_inner(&kf).expect("validating BASE");
+        let kf = kf_from_str(MIN_OVERRIDES)?;
```

Comment:
This one passed before because `COMPLEX` had every override, including e.g. `initramfs=true` etc.  This more targeted one specifically tests that we go to `baserefspec` when `remove` or `replace-local` are set.